supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/api/request/GcpBatchGroupedRequests.scala

```diff
@@ -0,0 +1,25 @@
+package cromwell.backend.google.batch.api.request
+
+import cromwell.backend.google.batch.api.BatchApiRequestManager._
+import cromwell.backend.google.batch.api.BatchApiResponse
+
+import scala.concurrent.{Future, Promise}
+import scala.util.Try
+
+// Mirrors com.google.api.client.googleapis.batch.BatchRequest but this is immutable
+class GcpBatchGroupedRequests(requests: List[(BatchApiRequest, Promise[Try[BatchApiResponse]])]) {
```

Comment:
Is there still a value in grouping them in that case, or should we just fire them off as they come in?

Reply:
Originally, this was created because we hoped that Google had an alternative to Batch requests, by now, Google has confirmed that there is no way to do that.

These are some notes from our internal discussions:
1. The code becomes way simpler if this grouping gets removed.
2. We have not checked the potential implications on creating a batch client for every request, or, reusing the same client for the application's lifecycle.
3. Grouping requests could allow us to eventually implement streaming like fs2/akka-stream, which could allow us to throttle the requests, still, if Cromwell already does this in another layer, this becomes unnecessary.

Given that the current code has been tested so many times, my suggestion would be to keep it and potentially remove it in another iteration.